src/main/scala/is/hail/expr/AST.scala

```diff
@@ -324,7 +324,7 @@ case class Const(posn: Position, value: Any, t: Type) extends AST(posn) {
   }
 }
 
-case class Select(posn: Position, lhs: AST, rhs: String) extends AST(posn, lhs) {
+case class SelectAST(posn: Position, lhs: AST, rhs: String) extends AST(posn, lhs) {
```

Comment:
Also, this is a straightforward change you could push independently and would get approved almost immediately.